docs/introduction.md

```diff
@@ -2,150 +2,449 @@
 
-```python title="clock01.py"
---8<-- "docs/examples/introduction/clock01.py"
+```sass 
+Stopwatch {
+    layout: horizontal;
+    background: $panel-darken-1;
+    height: 5;
+    padding: 1;
+    margin: 1;
+}
 ```
 
-Here's what you will see if you run this code:
+The first line tells Textual that the styles should apply to the `Stopwatch` widget. The lines between the curly brackets contain the styles themselves.
+
+Here's how this CSS code changes how the `Stopwatch` widget is displayed.
+
+<div class="excalidraw">
+--8<-- "docs/images/stopwatch_widgets.excalidraw.svg"
+</div>
 
-```{.textual path="docs/examples/introduction/clock01.py"}
+- `layout: horizontal` aligns child widgets horizontally from left to right.
+- `background: $panel-darken-1` sets the background color to `$panel-darken-1`. The `$` prefix picks a pre-defined color from the builtin theme. There are other ways to specify colors such as `"blue"` or `rgb(20,46,210)`.
+- `height: 5` sets the height of our widget to 5 lines of text.
+- `padding: 1` sets a padding of 1 cell around the child widgets.
+- `margin: 1` sets a margin of 1 cell around the Stopwatch widget to create a little space between widgets in the list.
 
+
+Here's the rest of `stopwatch03.css` which contains further declaration blocks:
+
+```sass
+TimeDisplay {
+    content-align: center middle;
+    opacity: 60%;
+    height: 3;
+}
+
+Button {
+    width: 16;    
+}
+
+#start {
+    dock: left;
+}
+
+#stop {
+    dock: left;
+    display: none;
+}
+
+#reset {
+    dock: right;
+}
 ```
 
-This script imports App as before, but also the `Widget` class from `textual.widget`, which is the base class for all Widgets. To create a Clock widget we extend from the Widget base class:
+The `TimeDisplay` block aligns text to the center (`content-align`), fades it slightly (`opacity`), and sets its height (`height`) to 3 lines.
+
+The `Button` block sets the width (`width`) of buttons to 16 cells (character widths).
 
-```python title="clock01.py" hl_lines="7 8 9 10 11 12 13"
---8<-- "docs/examples/introduction/clock01.py"
+The last 3 blocks have a slightly different format. When the declaration begins with a `#` then the styles will be applied to widgets with a matching "id" attribute. We've set an ID on the Button widgets we yielded in compose. For instance the first button has `id="start"` which matches `#start` in the CSS.
+
+The buttons have a `dock` style which aligns the widget to a given edge. The start and stop buttons are docked to the left edge, while the reset button is docked to the right edge.
+
+You may have noticed that the stop button (`#stop` in the CSS) has `display: none;`. This tells Textual to not show the button. We do this because we don't want to dsplay the stop button when the timer is *not* running. Similarly we don't want to show the start button when the timer is running. We will cover how to manage such dynamic user interfaces in the next section.
```

Comment:
```suggestion
You may have noticed that the stop button (`#stop` in the CSS) has `display: none;`. This tells Textual to not show the button. We do this because we don't want to display the stop button when the timer is *not* running. Similarly we don't want to show the start button when the timer is running. We will cover how to manage such dynamic user interfaces in the next section.
```